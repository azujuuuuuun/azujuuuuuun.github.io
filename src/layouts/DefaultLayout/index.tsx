import React from "react";

import Header from "../../components/Header";
import Footer from "../../components/Footer";
import styles from "./styles";

interface Props {
  isHome?: boolean;
  children: React.ReactNode;
}

const DefaultLayout: React.FC<Props> = ({ isHome, children }) => (
  <div style={styles.container}>
    <Header isHome={isHome} />
    {children}
    <Footer isHome={isHome} />
  </div>
);

export default DefaultLayout;
