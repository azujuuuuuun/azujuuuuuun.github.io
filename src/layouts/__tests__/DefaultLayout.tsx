import React from 'react';
import renderer from 'react-test-renderer';

import DefaultLayout from '../DefaultLayout';

describe('DefaultLayout', (): void => {
  it('renders correctly', (): void => {
    const tree = renderer.create(
      <DefaultLayout>
        <p>Hello, world!</p>
      </DefaultLayout>
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders isHome prop correctly', (): void => {
    const props = { isHome: true };
    const tree = renderer.create(
      <DefaultLayout {...props}>
        <p>Hello, world!</p>
      </DefaultLayout>
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
