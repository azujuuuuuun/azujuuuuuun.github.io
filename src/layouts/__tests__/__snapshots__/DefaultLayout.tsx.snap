// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DefaultLayout renders correctly 1`] = `
<div
  style={
    Object {
      "margin": "3rem auto",
      "maxWidth": 650,
      "padding": "0 1rem",
    }
  }
>
  <header>
    <a
      aria-current="page"
      className=""
      href="/"
      onClick={[Function]}
      onMouseEnter={[Function]}
      style={Object {}}
    >
      azujuuuuuun.github.io
    </a>
  </header>
  <p>
    Hello, world!
  </p>
  <footer
    style={
      Object {
        "paddingTop": 20,
      }
    }
  >
    <ul
      style={
        Object {
          "margin": 0,
        }
      }
    >
      <li
        style={
          Object {
            "display": "inline-block",
            "marginRight": 16,
          }
        }
      >
        <a
          aria-current="page"
          className=""
          href="/"
          onClick={[Function]}
          onMouseEnter={[Function]}
          style={Object {}}
        >
          ホーム
        </a>
      </li>
      <li
        style={
          Object {
            "display": "inline-block",
            "marginRight": 16,
          }
        }
      >
        <a
          href="/about"
          onClick={[Function]}
          onMouseEnter={[Function]}
        >
          自己紹介
        </a>
      </li>
      <li
        style={
          Object {
            "display": "inline-block",
            "marginRight": 16,
          }
        }
      >
        <a
          href="/favorites"
          onClick={[Function]}
          onMouseEnter={[Function]}
        >
          好きなもの
        </a>
      </li>
      <li
        style={
          Object {
            "display": "inline-block",
            "marginRight": 16,
          }
        }
      >
        <a
          href="/links"
          onClick={[Function]}
          onMouseEnter={[Function]}
        >
          リンク
        </a>
      </li>
    </ul>
    <small>
      ©Copyright jun. All rights reserved.
    </small>
  </footer>
</div>
`;

exports[`DefaultLayout renders isHome prop correctly 1`] = `
<div
  style={
    Object {
      "margin": "3rem auto",
      "maxWidth": 650,
      "padding": "0 1rem",
    }
  }
>
  <header>
    <h1>
      <a
        aria-current="page"
        className=""
        href="/"
        onClick={[Function]}
        onMouseEnter={[Function]}
        style={Object {}}
      >
        azujuuuuuun.github.io
      </a>
    </h1>
  </header>
  <p>
    Hello, world!
  </p>
  <footer
    style={
      Object {
        "paddingTop": 20,
      }
    }
  >
    <small>
      ©Copyright jun. All rights reserved.
    </small>
  </footer>
</div>
`;
